doozer/doozerlib/cli/release_gen_assembly.py

```diff
@@ -554,17 +553,14 @@ def _get_advisories_release_jira(self) -> Tuple[Dict[str, int], str]:
                 return advisories, release_jira
 
         previous_group = releases_config.releases[previous_assembly].assembly.group
-        advisories.update(previous_group.advisories.primitive())
+        previous_advisories = previous_group.advisories.primitive()
 
-        # prerelease advisory already associated with an assembly should be shipped, or dropped
-        # do not reuse it
-        # RCs can be designated as prerelease, but not always
-        # see https://docs.google.com/document/d/1S_ivD8Sh85LuZKnClqvoRvIhn4hNJQA76jpfUaFZjUA/edit
-        if 'prerelease' in advisories:
-            if self.pre_ga_mode == 'prerelease':
-                advisories['prerelease'] = -1
-            else:
-                advisories.pop('prerelease')
+        # preGA advisories (prerelease/advance) associated with an assembly should not be reused
+        # they should be shipped or dropped if not shipping
+        for key in preGA_advisory_type:
+            previous_advisories.pop(key, None)
+
+        advisories.update(previous_advisories)
```

Comment:
I don't think the update method removes keys from the dict its trying to update right?

Reply:
Yes removal is done in the previous line `.pop`

Reply:
`advisories` are initialized as 
```
advisories = {
            'image': -1,
            'rpm': -1,
            'extras': -1,
            'metadata': -1,
        }
```
so we're good